apfsprogs: 0-unstable-2024-09-27 -> 0.2.0

Luflosi opened the pull request:
https://github.com/linux-apfs/apfsprogs/releases/tag/v0.2.0

Also format with nixfmt-rfc-style

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Luflosi committed:
apfsprogs: format with nixfmt-rfc-style

Luflosi committed:
apfsprogs: 0-unstable-2024-09-27 -> 0.2.0

https://github.com/linux-apfs/apfsprogs/releases/tag/v0.2.0

Also add some simple tests to check that all tools can output the correct version.

FliegendeWurst commented:
@ofborg build apfsprogs

FliegendeWurst commented:
@ofborg build apfsprogs

FliegendeWurst commented on pkgs/by-name/ap/apfsprogs/package.nix:
```diff
@@ -1,29 +1,32 @@
-  '';
+  postPatch =
+    let
+      shortRev = builtins.substring 0 9 finalAttrs.src.rev;
```
This no longer makes sense, the src.rev is a tag.

Luflosi replied:
Good catch! Thanks for noticing.

Luflosi commented:
I fixed the malformed version being compiled into the program as pointed out in the review comment.
I also added some checks to test if each tool can output the correct version.

pbsds commented:
@ofborg build apfsprogs.passthru.tests

FliegendeWurst commented:
@ofborg build apfsprogs.passthru.tests
